### ISSUE TITLE
Remove hurdle of creating an account

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#1238]
Google Play statistics suggests that a lot of people uninstall the app just after installing it.
I suspect that the biggest blocker is the login screen. Most people probably give up at that step.

Proposed solution: Allow people to browse Commons, and even upload pictures, without the hurdle of choosing a username and password.

The app could show: Log in with existing account / Create an account / Continue without an account

"Continue without an account" would actually create a random name and password, and store them in the app's local database.
Uninstalling the app would mean the account is lost forever. Or we could have a small dialog that shows the username/password that have been created.

I am not sure whether all of this is possible (for instance account creation might require solving a captcha which might be difficult to make the user solve), but it might be worth considering.

**Comment by knight-shade** [comment]
@nicolas-raoul Smart Lock might be just the right thing for this AFAIK.
https://developers.google.com/identity/smartlock-passwords/case-studies

**Comment by misaochan** [comment]
I agree with allowing the user to browse Commons and view Nearby without having to be logged in. 

A few concerns though:
- I suspect doing this will require quite a thorough overhaul of many parts of the codebase? AFAIK our current (partially legacy).code is built on the assumption that the user is logged in. It will likely be a much larger task than it seems on the surface. 
- I'm not sure if allowing users to upload with throwaway accounts will be a popular addition from the perspective of Commons patrollers :)
- Not sure if Smart Lock is compatible with WMF privacy policies

**Comment by diddypod** [comment]
Allowing users who aren't logged in to upload images won't require a very drastic refactoring of the application, since MediaWiki and the APIs support anonymous edits (in a manner of speaking, it attributes it to the IP address of the user, instead).
However, the ContributionsActivity would have to go (for a user who isn't logged in, that is) , since the IP of a mobile user will keep changing.

Further, I don't think throwaway accounts are feasible, given the users need to fill in a captcha while signing up, and the point of this is to spare them that experience.

**Comment by nicolas-raoul** [comment]
As tipped by @sivaraam we should probably do without the login screen, and directly show featured images and browsing to the user.

Then when the user tries to upload a picture we should popup the login screen.
We could also have a "Login" item in the navigation menu for people who want to login without uploading an image.

**Comment by ujjwalagrawal17** [comment]
@nicolas-raoul @misaochan @neslihanturan @maskaravivek 
The order in which I think skip feature should work is Welcome Screen will come in first run after that everytime explore activity will come if user is logged in or not. The drawer menu will changed if user is not logged in

**Comment by neslihanturan** [comment]
My opinion:
After welcome screen, Explore screen should come if user is not logged in. Otherwise (if user is logged in), Main screen (which will be Upload and Nearby soon) should be visible. Since the main function of our app is uploading.

**Comment by nicolas-raoul** [comment]
Like Nes, I am in favour of keeping an upload-related activity (currently "My Uploads") as the default activity for logged-in users.
Everyone else what is your opinion?

**Comment by maskaravivek** [comment]
I tried to conduct an informal survey at Wikimania to understand what would they like to see as soon as they enter the app. The opinions were divided, with many of the people wanting My Uploads activity as their home activity and many others were hoping for some explore commons feature(For them seeing their own uploads was not very useful). The sample space here is 15-20 people. 

Given this dichotomy, we might consider giving this option to the users themselves. They could set the default landing page in settings. If Settings doesn't seem appropriate for this we can probably continue having My Uploads as the default activity for logged-in users as many people do find it useful. Personally, I as a user who doesn't upload many pictures would love to explore pictures uploaded by others.

**Comment by sivaraam** [comment]
> If Settings doesn't seem appropriate for this we can probably continue having My Uploads as the default activity for logged-in users as many people do find it useful.

I think giving the option to the user would result in us opening up the [pandora box](https://en.wikipedia.org/wiki/Pandora's_box). It might very likely cause unwanted issues in the upcoming days. 

> Personally, I as a user who doesn't upload many pictures would love to explore pictures uploaded by others.

Me too (regardless of whether I'm not logged in/not).

I don't think @neslihanturan's statement "Since the main function of our app is uploading." would hold once the browsing and Explore feature have been added to the app. I anticipate a lot of users would be using the app for browsing. So, the app wouldn't be upload only anymore. Anyways, that's an anticipation!

To be honest, being a very in-frequent Commons app uploader: I never found the usefulness of the contributions screen! It shows the images I've uploaded which might be useful. But I don't think it's as useful as the browsing or Explore feature (I'm not even able to download the images I upload! #1216). That's me, anyway.

That said, I would be in favor of the app choosing the app the main screen rather than letting the user. It would be nice if we could have a survey with a wider audience. 15-20 people seems **too** low. The results are quite likely to vary for a wider audience.

**Comment by nicolas-raoul** [comment]
I believe we should avoid proposing options unless really necessary, because:
- It makes important options less visible
- It multiplies the possibilities of bugs
- It makes testing (and trying to reproduce crashes) take more time
I asked our Facebook community to comment here.

**Comment by misaochan** [comment]
To answer @ujjwalagrawal17 's question specifically, I personally think that the best way to implement this for the time being is to: 
(1) display tutorial first, always, then 
(2) display login screen with a prominent option to "proceed without logging in". If they decide to do that, then 
(3a) we can display the browse activity as the main screen, otherwise 
(3b) if they login, we proceed to uploads as normal

**Comment by nicolas-raoul** [comment]
@misaochan About step (2): It was the topic of https://github.com/commons-app/apps-android-commons/issues/1238#issuecomment-399892860 in my understanding it has already been decided upon, and actually Ujjwal has implemented this way.

**Comment by misaochan** [comment]
@nicolas-raoul But in that case wouldn't that make the decision for us that browse must be the main activity?

**Comment by misaochan** [comment]
Hmm. In that case my only concern is that existing users would be surprised with the Browse screen and the lack of login screen (and since Browse currently looks a lot like Uploads, they may miss the title and wonder why they have such strange uploads now), but hopefully they will still try to upload anyway and be prompted to login.

It also raises the question of what we will do when #725 is implemented, which will be soon.

**Comment by ujjwalagrawal17** [comment]
Yup now thinking again I also agree with @misaochan. We should make it like she is telling so that the current users (they are mainly uploaders) don't get totally disturbed. We can maybe change it later.

**Comment by nicolas-raoul** [comment]
We don't have _that_ many users right now, so I think it is OK to prioritize future users' experience :-)
@ujjwalagrawal17 don't you think it is a bit late to change that again? If you really feel that way please do like you want, and let us know as soon as it is ready to test.

**Comment by ujjwalagrawal17** [comment]
Wikipedia app also directly shows explore page but they have a nice card for first time users which tells the advantages of login. One of the aim of Wikimedia Commons is to provide access to free usable images to everyone. Not just uploading. @misaochan if we delay it now then maybe it can be delayed for a long time. So I think we can skip the step you are saying and directly show explore as the main page for not logged in users.

**Comment by misaochan** [comment]
IMO there is a big difference between the main aim of the Wikipedia app and the (current) main aim of the Commons app. The Wikipedia app is predominantly for reading, not editing. The Commons app is (to current users) mainly for uploading, not browsing. Things could certainly change in the future depending on the response to the Browse feature, but at the moment I really think we are not ready for a switch. At the very least, if we must have a switch, we should have a demo to walk users through it (here is the browse feature, you can still log in and upload here, when you log in you will be brought to your uploads, etc), which would take more time to implement IMO.

@nicolas-raoul Given that we don't know whether users actually prefer Browse or Uploads as the main UI, what is the benefit to future users of immediately having a browse screen instead of a "proceed without logging in" option? It seems to me that both would be equally useful for future users, whereas the latter would be more comfortable for existing users, so the net gain is higher.

I understand GSoC time constraints, but this is not a mandatory GSoC task, right? So Ujjwal can work on it as he likes, and if it's not ready by the time GSoC is over, there should be no problems?

**Comment by nicolas-raoul** [comment]
With "Login" and "Upload" items in the navbar, I am sure existing users would not be lost.

I have no strong preference between the two, actually. It is just that there was a consensus and Ujjwal implemented it, so I feel bad about going back on this. But if Ujjwal is OK to implement as you say, I am totally OK with it :-)

Allowing non-logged-in users to browse is a mandatory GSoC task, scheduled "July 24 to July 30" at https://phabricator.wikimedia.org/T188597

**Comment by misaochan** [comment]
Oh, okay! I guess if time is really an issue he can submit the PR first and we can alter the login flow later as needed. There is no GSoC requirement for it to actually be released to users, right? So we can release the current state of Browse first, and then leave the activity switch to a future decision.

**Comment by nicolas-raoul** [comment]
Understood!

So, @ujjwalagrawal17 please decide whether you prefer to first show a login dialog or not, and implement it :-) Please tell us what you choose and then let us know when it is ready.

**Comment by ujjwalagrawal17** [comment]
We can do it as @misaochan is saying it. I will update the PR accordingly and let you know.
